### DISCUSSION TITLE
Row Level Security Policy Supabase

### DISCUSSION THREAD

**Discussion by boogerjosh** [supabase/supabase#16305] [Questions] [answered]
### Does anyone know how to solve this?
I get 401 for 'new row violates row-level security policy for table "contactus"'
<img width="221" alt="Screenshot 2023-08-07 at 23 10 50" src="https://github.com/supabase/supabase/assets/94213206/e0777d16-95bb-4b1c-8fa9-4c5f8bee998d">

I want to insert a data without authentication login or something, All user can insert data with just connecting to database. I tried to write the RLS like this
<img width="1166" alt="Screenshot 2023-08-07 at 23 13 06" src="https://github.com/supabase/supabase/assets/94213206/088b67d9-0549-417d-befe-2852954e18ef">

And in my code in client i write it like this:
<img width="476" alt="Screenshot 2023-08-07 at 23 14 11" src="https://github.com/supabase/supabase/assets/94213206/794a8712-d2c6-4f41-8966-83aafbd27d34">
s:
<img width="400" alt="Screenshot 2023-08-07 at 23 15 37" src="https://github.com/supabase/supabase/assets/94213206/c58a9a6c-1177-486e-974a-35cc4edd2b61">

Is there something should i fix or add to make it work? Help me please:)

**Comment by GaryAustin1** [comment, marked as answer]
You also need to meet select policy because your .insert has select on it.

**Reply by boogerjosh** [reply]
Can you show me the example for that?

**Reply by GaryAustin1** [reply]
You just add a select RLS policy to the table like you did for insert.
![image](https://github.com/supabase/supabase/assets/54564956/809b3927-4545-4dc2-acac-f78e7ae32e38)
This does mean anyone can read your files.
Your other option is to drop the .select() on the insert call.

**Reply by boogerjosh** [reply]
It works! Thanks!
